batch/batch/front_end/front_end.py

```diff
@@ -1158,18 +1176,25 @@ async def insert(tx):
 
         bp_cost_record = await tx.execute_and_fetchone(
             '''
-SELECT billing_projects.msec_mcpu, COALESCE(SUM(`usage` * rate), 0) AS cost
+SELECT billing_project, COALESCE(SUM(`usage` * rate), 0) AS cost
 FROM billing_projects
-INNER JOIN aggregated_billing_project_resources
-  ON billing_projects.name = aggregated_billing_project_resources.billing_project
-INNER JOIN resources
-  ON resources.resource = aggregated_billing_project_resources.resource
+LEFT JOIN (
+  SELECT billing_project, resource_id, CAST(COALESCE(SUM(`usage`), 0) AS SIGNED) AS `usage`
+  FROM billing_projects
+  LEFT JOIN aggregated_billing_project_user_resources_v2
+    ON billing_projects.name = aggregated_billing_project_user_resources_v2.billing_project
+  WHERE billing_projects.name = %s
+  GROUP BY aggregated_billing_project_user_resources_v2.billing_project, aggregated_billing_project_user_resources_v2.resource_id
+) AS usage_t ON usage_t.billing_project = billing_projects.name
+LEFT JOIN resources
+  ON resources.resource_id = usage_t.resource_id
 WHERE billing_projects.name = %s
+GROUP BY usage_t.billing_project;
```

Comment:
Are the two `WHERE billing_projects.name = %s` here redundant? Might not matter but could we get rid of the outer WHERE statement by doing an inner join with the table produced by the subquery?

Reply:
I'm not sure if we can guarantee order of how things are evaluated. This is a question for @danking

Reply:
I agree that an inner join avoids the need for a where, but it's not obvious to me that the optimizer is smart enough to propagate the where condition up to the outer-most inner join.

Hmm. Why are we joining and grouping by billing project if we're actually computing a point query? Seems like we should just:

```sql
select coalesce(sum(`usage` * rate)) as cost
from (
  select resource_id, cast(coalesce(sum(`usage`), 0) AS SIGNED) as `usage`
  from aggregated_billing_project_user_resources_v2
  left join resources on resources.resource_id = aggregated_billing_project_user_resources_v2.resource_id
  where billing_project = %s
  group by resource_id
)
```

Reply:
You're right. I was over complicating this.